stats/view/worker.go

```diff
@@ -143,9 +213,15 @@ func newWorker() *worker {
 		c:          make(chan command, 1024),
 		quit:       make(chan bool),
 		done:       make(chan bool),
+
+		exporters: make(map[Exporter]struct{}),
 	}
 }
 
+func (w *worker) Start() {
+	go w.start()
```

Comment:
There is another way to export data instead of exporting ViewData. This is [exporting metrics](https://github.com/census-instrumentation/opencensus-go/tree/master/metric/metricexport)  using Reader and IntervalReader. The Reader/IntervalReader reads metrics from all producer. Worker here is a producer. The existing w.start() method adds the worker (w) as producer. This means all workers will be producer and metrics from each producer will be available to the exporter using [ExportMetrics](https://github.com/census-instrumentation/opencensus-go/blob/d835ff86be02193d324330acdb7d65546b05f814/metric/metricexport/export.go#L25) interface. You are probably not using that interface. Otherwise this will not work.
Alternatively, you could register a wrapper exporter that creates SD exporter with specific resource. The wrapper exporter filters the time-series based on labels. It may create extra processing overhead but will limit the code change in knative.

Reply:
It looks like using ExportMetrics requires tracking and aggregating all the metrics in something (which would basically be `views`).

Another option here would be to [add an additional internal key](https://github.com/census-instrumentation/opencensus-go/blob/master/stats/view/worker_commands.go#L164) to `worker.measure` that keyed off the Resource. Unfortunately, that looks to require some additional surgery to `encodeWithKeys` and other parts of the general code path that I was hoping to avoid.